Regression_CF.R

```diff
@@ -0,0 +1,69 @@
+#setwd("C:/Users/frase/Documents/Rclass/Regression-repo")
+
+read.csv("iris.csv")
+
+#rename vectors
+Slength <- iris$Sepal.Length
+Plength <- iris$Petal.Length
+
+x <- Slength
+y <- Plength
+lengths.lm <- lm(y~x) 
+coef(lengths.lm)
+b <- coef(lengths.lm)
+int <- b[1]
+slope <- b[2]
+
+plotwithreg <- function(x=Slength, y=Plength, title="Sepal and Petal Morph", cxlab="Sepal Length", cylab="Petal Length", ccol=iris$Species) {
+	plot(x, y, pch=10, col=ccol, main=title, xlab=cxlab, ylab=cylab)
+	abline(int, slope, col="blue")
+
+	yhat <- slope*x + int
+	points(x, yhat)
+	segments(x, y, x, yhat)
+
+	error <- yhat - y #error in y from actual point to point on regression line
+	error^2
+	sse <- sum(error^2)
+	sse #obtain sum of squared errors
+
+	text(7.5,2, paste("sse=", round(sse, 2)))
+	text(7.5,1.5, paste("slope=", round(slope, 2)))
+	text(7.5,1, paste("intercept=", round(int, 2)))
+	legend("topleft", legend=levels(iris$Species), col= c(1,2,3), pch=10)
+}
+
+#plot with default values
+plotwithreg()
+#set your x, y, and labels
+plotwithreg(Slength, Plength, "Sepal and Petal Morph", "Sepal Length", "Petal Length")
+
+#Try it with arbitrary slopes and intercepts
+slope = 2
+plotwithreg(Slength, Plength, "Sepal and Petal Morph", "Sepal Length", "Petal Length")
```

Comment:
Instead of changing the value of slope in the global environment, you should be passing slope into the function, so that you can change the value of the slope within the function.